packages/angular_devkit/build_angular/src/tools/esbuild/stylesheets/sass-language.ts

```diff
@@ -97,31 +103,49 @@ async function compileString(
             url,
             options: FileImporterWithRequestContextOptions,
           ): Promise<URL | null> => {
+            let resolved = resolutionCache.get(url);
+            if (resolved !== undefined) {
+              return resolved;
+            }
+
+            // Default to not found
+            resolved = null;
+
             const result = await resolveUrl(url, options);
             if (result.path) {
-              return pathToFileURL(result.path);
+              resolved = pathToFileURL(result.path);
+            } else {
+              // Check for package deep imports
+              const parts = url.split('/');
+              const hasScope = parts.length >= 2 && parts[0].startsWith('@');
+              const [nameOrScope, nameOrFirstPath, ...pathPart] = parts;
+              const packageName = hasScope ? `${nameOrScope}/${nameOrFirstPath}` : nameOrScope;
```

Comment:
Suggestion: Refactoring to a `parsePackageName` function would be useful to isolate the specifics of package parsing. Not sure if that's feasible with the following `join`, but it would help a lot with readability if we can reduce the responsibility here.